plugins/modules/configuration_item.py

```diff
@@ -0,0 +1,406 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: configuration_item
+
+author:
+  - Manca Bizjak (@mancabizjak)
+  - Miha Dolinar (@mdolin)
+  - Tadej Borovsak (@tadeboro)
+
+short_description: Manage ServiceNow configuration items
+
+description:
+  - Create, delete or update a ServiceNow configuration item.
+  - For more information, refer to the ServiceNow configuration management documentation at
+    U(https://docs.servicenow.com/bundle/paris-servicenow-platform/page/product/configuration-management/reference/cmdb-table-property-descriptions.html).
+
+extends_documentation_fragment:
+  - servicenow.itsm.instance
+  - servicenow.itsm.sys_id
+
+seealso:
+  - module: servicenow.itsm.configuration_item_info
+
+options:
+  state:
+    description:
+      - State of the configuration item.
+    type: str
+    choices: [ present, absent ]
+    default: present
+  name:
+    description:
+      - The name of the configuration item.
+      - Required if the configuration item does not yet exist.
+    type: str
+  short_description:
+    description:
+      - Short description of the configuration item.
+    type: str
+  sys_class_name:
```

Comment:
What would be the right behavior if the user is trying to modify an existing item and specifies a different? Right now, we change the class, but if this operation can delete user data (for example, when changing from some more specific class down to `cmdb_ci`), I would say we should do something to warn the user (or even abort the play).

Maybe we should only allow setting the class when creating an item and make sure that modifications do not alter the class field?

Anyhow, this is not of critical importance and can be updated in a follow-up pull request if we decide to change it.